File: Claude/Intermediárias/sabbachard/Sabbachard.java
import java.util.Arrays;

public class Sabacchard {
    public static int maxScore(int[] deck) {
        int n = deck.length;
        int[][] dp = new int[n][n];
        
        for (int length = 1; length <= n; length++) {
            for (int start = 0; start <= n - length; start++) {
                int end = start + length - 1;
                if (length == 1) {
                    dp[start][end] = deck[start];
                } else {
                    dp[start][end] = Math.max(deck[start] + dp[start+1][end], 
                                              deck[end] + dp[start][end-1]);
                }
            }
        }
        
        return dp[0][n-1];
    }

    public static void main(String[] args) {
        int[] deck = {10, 19, 1, 2};
        System.out.println("Maximum score: " + maxScore(deck));
    }
}

File: Gemini/Intermediárias/sabbachard/Sabbachard.py
def max_score(deck):
  """Calculates the maximum possible score for a given deck of cards.

  Args:
    deck: A list of integers representing the cards in the deck.

  Returns:
    The maximum possible score.
  """

  # Base case: if there is only one card left, return its value
  if len(deck) == 1:
    return deck[0]

  # Calculate the maximum score by considering both options:
  # taking the first card or taking the last card
  score_first = deck[0] + max_score(deck[1:])
  score_last = deck[-1] + max_score(deck[:-1])

  # Return the maximum of the two scores
  return max(score_first, score_last)

# Example usage
deck = [10, 19, 1, 2]
print(max_score(deck))  # Output: 21